blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -37,36 +37,42 @@
 public class Tracker implements TrackerMBean {
 
     private static final Logger log = LoggerFactory.getLogger(Tracker.class);
-    private final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
+    private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
     private final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
 
     // Tracker is a singleton
-    private static final Tracker instance = new Tracker();
+    private static Tracker instance;
+    private static boolean isRegistered = false;
 
     private Set tenantIds = new HashSet();
     private boolean isTrackingDelayedMetrics = false;
     private Set<String> metricNames = new HashSet<String>();
 
     // private constructor for singleton
-    private Tracker(){
-        registerMBean();
-    }
+    private Tracker(){}
+
+    public static synchronized void register() {
+        if (instance == null) instance = new Tracker();
 
-    private void registerMBean() {
-        try {
-            ObjectName objectName = new ObjectName(trackerName);
+        if (!isRegistered) {
+            try {
+                ObjectName objectName = new ObjectName(trackerName);
 
-            // register TrackerMBean only if not already registered
-            MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
-            if (!mBeanServer.isRegistered(objectName)) {
-                ManagementFactory.getPlatformMBeanServer().registerMBean(this, objectName);
+                // register TrackerMBean only if not already registered
+                MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+                if (!mBeanServer.isRegistered(objectName)) {
+                    ManagementFactory.getPlatformMBeanServer().registerMBean(instance, objectName);
+                }
+            } catch (Exception exc) {
+                log.error("Unable to register mbean for " + instance.getClass().getSimpleName(), exc);
             }
-        } catch (Exception exc) {
-            log.error("Unable to register mbean for " + this.getClass().getSimpleName(), exc);
+
+            isRegistered = true;
         }
     }
 
     public static Tracker getInstance() {
```

Comment:
Do we need this method? It seems like people should just call Tracker.register() ?